ui/components/app/simulation-preview/asset-pill.tsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { Text } from '../../component-library';
+import {
+  AlignItems,
+  BackgroundColor,
+  BorderRadius,
+  Display,
+  FlexDirection,
+  TextVariant,
+} from '../../../helpers/constants/design-system';
+import { AssetInfo } from './types';
+
+const EthAssetPill: React.FC = () => (
+  <Text
+    display={Display.Flex}
+    flexDirection={FlexDirection.Row}
+    borderRadius={BorderRadius.pill}
+    alignItems={AlignItems.center}
+    backgroundColor={BackgroundColor.backgroundAlternative}
+    gap={1}
+    style={{
+      padding: '0px 8px 0 4px',
+    }}
+    variant={TextVariant.bodyMd}
+  >
+    <img alt="ETH logo" src="./images/eth_badge.svg" width="18px" />
+    ETH
+  </Text>
+);
+
+/**
+ * Displays a pill with an asset's symbol and name.
+ *
+ * @param props
+ * @param props.assetInfo
+ */
+export const AssetPill: React.FC<{ assetInfo: AssetInfo }> = ({
+  assetInfo,
+}) => {
+  if (assetInfo.isNative) {
+    return <EthAssetPill />;
+  }
+  return <Text variant={TextVariant.bodyMd}>Token</Text>;
```

Comment:
As we're only processing the native balance change from the original data, should we return `null` in this PR so it's explicit we're only supporting native changes at this stage?